pkg/writers/buffered_file_writer/bufferedfilewriter.go

```diff
@@ -8,29 +8,61 @@ import (
 	"io"
 	"os"
 	"sync"
+	"time"
 
 	"github.com/trufflesecurity/trufflehog/v3/pkg/cleantemp"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+type bufferPoolMetrics struct{}
+
+func (b *bufferPoolMetrics) recordGrowth(growthAmount int) {
+	growCount.Inc()
+	growAmount.Add(float64(growthAmount))
+}
+
+func (b *bufferPoolMetrics) recordShrink(amount int) {
+	shrinkCount.Inc()
+	shrinkAmount.Add(float64(amount))
+}
+
+func (b *bufferPoolMetrics) recordCheckoutDuration(duration time.Duration) {
+	checkoutDuration.Observe(float64(duration.Microseconds()))
+	checkoutCount.Inc()
+	checkoutDurationTotal.Add(float64(duration.Microseconds()))
+}
+
+// recordBufferRetrival groups the metrics updates for when a buffer is fetched.
```

Comment:
Why does this one get a comment but not the others?

Reply:
it's special. I'll add them across, i probably just got lazy 😞

Reply:
err actually they are pretty obvious i'll remove the comment as it seems redundant here. Happy to add them if you feel it adds value.